### PR TITLE
Remove deprecated code in `kornia.augmentation`

### PR DESCRIPTION
**Pull request by johnnv1**
Related to #1990 -- this patch must have removed all deprecated code in `kornia.augmentation`

#### Type of change
- [x] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/augmentation/random_generator/_3d/affine.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 from torch.distributions import Uniform
 
 from kornia.augmentation.random_generator.base import RandomGeneratorBase
-from kornia.augmentation.utils import _adapted_rsampling, _adapted_uniform, _singular_range_check, _tuple_range_reader
-from kornia.utils.helpers import _deprecated, _extract_device_dtype
+from kornia.augmentation.utils import _adapted_rsampling, _singular_range_check, _tuple_range_reader
+from kornia.utils.helpers import _extract_device_dtype
 
 
 class AffineGenerator3D(RandomGeneratorBase):
```
Comment (johnnv1) [inline kornia/augmentation/random_generator/_3d/affine.py:11]: Shouldn't these classes have "Random" in their name?

Comment (edgarriba) [reply kornia/augmentation/random_generator/_3d/affine.py:11]: Possibly

**Comment by edgarriba** [comment]
Any idea why is crashing?

**Comment by johnnv1** [comment]
> Any idea why is crashing?

No :/

**Comment by johnnv1** [comment]
I had a better look at it, and seems that the issue is related with `uniform.rsample()` when considered different `dtypes` leads to different results even when using the same seed

----

For example, at pdb in TestRandomAffineGen3D forward:

dtype32
```
(Pdb) angles
tensor([[18.8227, 13.8286, 13.9045],
        [19.1500, 19.5931, 16.0090]])
```

dtype64:
```
(Pdb) angles
tensor([[10.5815, 11.2359, 15.2617],
        [10.6291, 10.5258, 14.7678]], dtype=torch.float64)
```

----

minimal example:
```python
import torch
from torch.distributions import Uniform

ex16 = torch.tensor([[10., 20.],], dtype=torch.float16)
ex32 = torch.tensor([[10., 20.],], dtype=torch.float32)
ex64 = torch.tensor([[10., 20.],], dtype=torch.float64)

s16 = Uniform(ex16[0][0], ex16[0][1], validate_args=False)
s32 = Uniform(ex32[0][0], ex32[0][1], validate_args=False)
s64 = Uniform(ex64[0][0], ex64[0][1], validate_args=False)

print(s16,'\n', '-'*25)
print(s32,'\n', '-'*25)
print(s64,'\n', '-'*25)

print('-'*79)

torch.manual_seed(42)
print(s16.rsample((2,)),'\n', '-'*25)

torch.manual_seed(42)
print(s32.rsample((2,)),'\n', '-'*25)

torch.manual_seed(42)
print(s64.rsample((2,)),'\n', '-'*25)
```
```output
Uniform(low: 10.0, high: 20.0) 
 -------------------------
Uniform(low: 10.0, high: 20.0) 
 -------------------------
Uniform(low: 10.0, high: 20.0) 
 -------------------------
-------------------------------------------------------------------------------
tensor([15.5000, 17.1250], dtype=torch.float16) 
 -------------------------
tensor([18.8227, 19.1500]) 
 -------------------------
tensor([10.5815, 10.6291], dtype=torch.float64) 
```

**Comment by shijianjian** [comment]
It is. I implemented internal device and dtypes in the base class for all augmentations to guarantee the reproducibility.

https://github.com/kornia/kornia/blob/c8e54295dd8a96707478fbb1e8e7fdef05b9df79/kornia/augmentation/base.py#L62

Generally, you may think all the random generators will generate numbers from float32, then convert to float64 or so.


> I had a better look at it, and seems that the issue is related with `uniform.rsample()` when considered different `dtypes` leads to different results even when using the same seed
> 
> For example, at pdb in TestRandomAffineGen3D forward:
> 
> dtype32
> 
> ```
> (Pdb) angles
> tensor([[18.8227, 13.8286, 13.9045],
>         [19.1500, 19.5931, 16.0090]])
> ```
> 
> dtype64:
> 
> ```
> (Pdb) angles
> tensor([[10.5815, 11.2359, 15.2617],
>         [10.6291, 10.5258, 14.7678]], dtype=torch.float64)
> ```
> 
> minimal example:
> 
> ```python
> import torch
> from torch.distributions import Uniform
> 
> ex16 = torch.tensor([[10., 20.],], dtype=torch.float16)
> ex32 = torch.tensor([[10., 20.],], dtype=torch.float32)
> ex64 = torch.tensor([[10., 20.],], dtype=torch.float64)
> 
> s16 = Uniform(ex16[0][0], ex16[0][1], validate_args=False)
> s32 = Uniform(ex32[0][0], ex32[0][1], validate_args=False)
> s64 = Uniform(ex64[0][0], ex64[0][1], validate_args=False)
> 
> print(s16,'\n', '-'*25)
> print(s32,'\n', '-'*25)
> print(s64,'\n', '-'*25)
> 
> print('-'*79)
> 
> torch.manual_seed(42)
> print(s16.rsample((2,)),'\n', '-'*25)
> 
> torch.manual_seed(42)
> print(s32.rsample((2,)),'\n', '-'*25)
> 
> torch.manual_seed(42)
> print(s64.rsample((2,)),'\n', '-'*25)
> ```
> 
>  ```
> Uniform(low: 10.0, high: 20.0) 
>  -------------------------
> Uniform(low: 10.0, high: 20.0) 
>  -------------------------
> Uniform(low: 10.0, high: 20.0) 
>  -------------------------
> -------------------------------------------------------------------------------
> tensor([15.5000, 17.1250], dtype=torch.float16) 
>  -------------------------
> tensor([18.8227, 19.1500]) 
>  -------------------------
> tensor([10.5815, 10.6291], dtype=torch.float64) 
> ```

**Comment by johnnv1** [comment]
So, what is the solution here? Add different expected results for each dtype?

**Comment by shijianjian** [comment]
@johnnv1 It would be the best if you can fix the problems for 3D generators. I think the internal dtype conversion is somehow failed.

**Comment by johnnv1** [comment]
to be able to investigate this, what would be the expected behavior -- generate the parameters in float32 and then cast to dtype, or generate based on dtype?

**Comment by edgarriba** [comment]
I think most scalable approach for now is to generate in fp32

File: kornia/augmentation/utils/param_validation.py [hunk 2]
```diff
@@ -94,83 +96,78 @@ def _singular_range_check(
...
     tensor([[0.2000, 0.3000],
             [0.2000, 0.3000],
             [0.2000, 0.3000]])
     """
     target_shape = torch.Size([target_size, 2])
-    if not torch.is_tensor(input_range):
-        if isinstance(input_range, (float, int)):
+
+    if isinstance(input_range, Tensor):
+        # TODO: Investigate if we should keeping ignoring the `device` argument when `input_range` is a Tensor
```
Comment (johnnv1) [inline kornia/augmentation/utils/param_validation.py:116]: Just to point out here, the problem was with this parameter validation function, which was not considering the `dtype` argument when the input is a tensor. Currently, it continues to ignore the `device` parameter if it is a tensor, otherwise it causes `Expected all tensors to be on the same device`

Comment (johnnv1) [reply kornia/augmentation/utils/param_validation.py:116]: Maybe the right thing here, should be considering the device argument, and after generate the parameter move to the desired device?

**Comment by johnnv1** [comment]
the CI failing now are the well-known problem related to #2027
